programs/mpl-core/src/plugins/lifecycle_hook.rs

```diff
@@ -0,0 +1,22 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use super::{ExternalPluginSchema, ExtraAccount};
+
+/// Lifecycle hook that CPIs into a hooked program specified in the `ExternalPluginKey`.  This hook
+/// is used for any lifecycle events that were selected in the `ExternalPluginRecord` for the
+/// plugin.  If any extra accounts are present in the `extra_accounts` optional Vec, then these
+/// accounts are added to the CPI call in the order in which they are in the Vec.  Any PDAs in the
+/// Vec are derived using the hooked program.  The hooked program will return a validation result
+/// and new data to store at the plugin's data offset (which in the account is immediately after
+/// this header).
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, Eq, PartialEq)]
+pub struct LifecycleHook {
+    /// The extra accounts to use for the lifecycle hook.
+    pub extra_accounts: Option<Vec<ExtraAccount>>,
+    /// Schema for the data used by the plugin.
+    pub schema: Option<ExternalPluginSchema>,
```

Comment:
Can we just have a default schema rather than making this an option?

Reply:
yeah I think you are right and it should default to like raw binary if they don't specify anything

Reply:
done

Reply:
Updated.  As part of this change I also just created `Init` and `Update` structs for each external plugin so that trying to reuse the plugin for initial specification or updating it wouldn't conflict with representation we actually want to store.  See https://github.com/metaplex-foundation/mpl-core/pull/58/commits/176c8af197e4fea8cd1c76141364a2487fe95820 for more info.